docs/DeveloperGuide.md

```diff
@@ -1,37 +1,351 @@
+
+
+
+#### Sequence Diagram
+
+The following sequence diagram illustrates how `Parser` parse command for `AddTaskCommand`:
+
+![Parse_Add_Task_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/ParseAddTaskClassDiagram.png)
+
+A closer look on how `AddTaskCommand` is executed is shown below:
+
+![Add_Task_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/AddTaskClassDiagram.png)
+
+#### Implementation considerations:
+The `AddTaskCommand` has been implemented with a design that prioritizes scalability and maintainability. This allows the system to easily support additional task types in the future without requiring major code changes. For example, if new task types are needed in the future (e.g., `event`, `appointment`), they can be added by simply extending the Task class hierarchy and updating the createTask() factory method.
+### **FindCommand**
+
+The find feature allows users to find the name of a patient, or the name of a task. This is facilitated by the `FindPatientCommand` and `FindTaskCommand`.
+
+1. **User Input**: The user enters the `find` command followed by keywords
+2. **Command Parsing**: The `Parser` parses the input and creates a `FindParser` object.
+3. **Execution**: The `FindPatientCommand` or `FindTaskCommand` searches for the task.
+
+#### Sequence Diagram
+
+The following sequence diagram illustrates how the `FindPatientCommand` is executed:
+
+![Find_Sequence_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/FindSequenceDiagram.png)
+
+### **Completion Rate Feature**
+
+The completion rate feature provides users with task progress summaries for individual patients and across all patients in the hospital. It consists of two main commands:
+
+1. **Show Task List for a Patient**: Displays a specific patient’s tasks with the percentage completed.
+        - The user initiates `showTaskList()`.
+        - `Ui` retrieves the selected `Patient` from `Hospital`, then calls `getTaskList()` to access tasks.
+        - The `TaskList` calculates the completion rate and returns it to `Ui`.
+        - `Ui` formats and displays the task list and completion rate to the user.
+
+2. **Show Patient List with Completion Rate**: Displays all patients with their overall task completion rates.
+        - The user initiates `showPatientListWithCompletionRate()`.
+        - `Ui` calls `calculateOverallCompletionRate()` on `Hospital`, which iterates through each `Patient` to get individual completion rates from their `TaskList`.
+        - `Hospital` aggregates these to determine the overall completion rate, then returns the list of patients with completion data.
+        - `Ui` formats and displays the list to the user.
+
+#### Sequence Diagram
+
+The following sequence diagram illustrates the **Completion Rate Feature**:
+
+![Completion Rate Sequence Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/CompletionRateSequenceDiagram.png)
```

Comment:
If the two headings "Show task list for Patient" and "show patient list with complete rate" are two separate scenarios, then could consider splitting the sequence diagram into two for readability